apps/token/src/routes/governance/components/proposal-detail-header/proposal-header.tsx

```diff
@@ -0,0 +1,104 @@
+import { useTranslation } from 'react-i18next';
+import { Lozenge } from '@vegaprotocol/ui-toolkit';
+import type { ReactNode } from 'react';
+import type { Proposals_proposals } from '../../proposals/__generated__/Proposals';
+
+export const ProposalHeader = ({
+  proposal,
+}: {
+  proposal: Proposals_proposals;
+}) => {
+  const { t } = useTranslation();
+  const { change } = proposal.terms;
+
+  let headerText: string;
+  let detailsOne: ReactNode;
+  let detailsTwo: ReactNode;
+
+  switch (change.__typename) {
+    case 'NewMarket': {
+      headerText = `${t('New market')}: ${change.instrument.name}`;
+      detailsOne = (
+        <>
+          {t('Code')}: {change.instrument.code}.{' '}
+          {change.instrument.futureProduct?.settlementAsset.symbol ? (
+            <>
+              <span className="font-semibold">
+                {change.instrument.futureProduct.settlementAsset.symbol}
+              </span>{' '}
+              {t('settled future')}.
+            </>
+          ) : (
+            ''
+          )}
+        </>
+      );
+      break;
+    }
+    case 'UpdateMarket': {
+      headerText = `${t('Market change')}: ${change.marketId}`;
+      break;
+    }
+    case 'NewAsset': {
+      headerText = `${t('New asset')}: ${change.name}`;
+      detailsOne = (
+        <>
+          {t('Symbol')}: {change.symbol}.{' '}
+          <Lozenge>
+            {change.source.__typename === 'ERC20'
+              ? `ERC20 ${change.source.contractAddress}`
+              : `${t('Max faucet amount mint')}: ${
+                  change.source.maxFaucetAmountMint
+                }`}
+          </Lozenge>
+        </>
+      );
+      break;
+    }
+    case 'UpdateNetworkParameter': {
+      headerText = `${t('Network parameter')}`;
+      detailsOne = (
+        <>
+          <Lozenge>{change.networkParameter.key}</Lozenge> {t('to')}{' '}
+          <Lozenge>{change.networkParameter.value}</Lozenge>
+        </>
+      );
+      break;
+    }
+    case 'NewFreeform': {
+      const description = proposal.rationale.description.trim();
```

Comment:
question (blocking): Why all this logic here? Might be missing something but is this just to truncate text? I'd prefer to keep it simple and just rely on `text-overflow: ellipsis` if we can